lib/elixir/lib/record.ex

```diff
@@ -309,7 +317,9 @@ defmodule Record do
 
   # Callback invoked from record/0 and record/1 macros.
   @doc false
-  def __access__(atom, fields, args, caller) do
+  @spec __access__(tag, fields :: Keyword.t, args :: atom, caller :: Macro.Env) :: non_neg_integer
+  @spec __access__(tag, fields :: Keyword.t, args :: Keyword.t, caller :: Macro.Env) :: Macro.t
```

Comment:
I agree with you Andrea,
I have left only `arg :: ` as it it can have multiple types. Let me know if that's reasonable for you.